client/client_session.cpp

```diff
@@ -44,6 +44,16 @@ void nibaclient::client_session::handle_cmd(const std::string &input) {
         }
         std::vector<std::string> results;
         boost::split(results, input, boost::is_any_of("\t "));
+        if (results.empty())
```

Comment:
This makes sense. I'm curious why wasn't it here and what made you put it here now.

Reply:
because the size of results are actually matched before retrieving the arguments, but for dynamically sized result we kinda need to make sure it's not empty before accessing `results[0]`. (or we could just do `results.at(i)` and let it throw an exception)